### PR TITLE
Replace spring oauth with spring security

### PR DESCRIPTION
**Pull request by IvanBorislavovDimitrov**
LMCROSSITXSADEPLOY-2359

### CODE REVIEW

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java [hunk 1]
```diff
@@ -2972,16 +2973,16 @@ public CloudControllerRestClientHttpRequestFactory(ClientHttpRequestFactory dele
...
+//            String authorizationHeader = oauthClient.getAuthorizationHeader();
+//            if (authorizationHeader != null) {
+//                request.getHeaders()
+//                       .add(AUTHORIZATION_HEADER_KEY, authorizationHeader);
+//            }
+//
+//            if (cloudCredentials != null && cloudCredentials.getProxyUser() != null) {
+//                request.getHeaders()
+//                       .add(PROXY_USER_HEADER_KEY, cloudCredentials.getProxyUser());
+//            }
```
Comment (radito3) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java:2985]: same here

**Review by radito3 — COMMENTED** [review]
Nice

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/CloudControllerClient.java [hunk 2]
```diff
@@ -685,17 +684,6 @@ public void createApplication(String appName, Staging staging, Integer disk, Int
...
-    /**
-     * Stream recent log entries.
-     *
-     * Stream logs that were recently produced for an app.
-     * @param appName the name of the application
-     * @param since the start of the time frame for the logs
-     * @param till the end of the time frame for the logs
-     * @return the list of recent log entries
-     */
-    List<ApplicationLog> getRecentLogs(String appName, Date since, Date till);
```
[the remark this answers is not in the document]
Comment (IvanBorislavovDimitrov) [reply cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/CloudControllerClient.java:697]: Reverted, there are issues with this implementation and it breaks the xs2 interface

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/oauth2/Oauth2AccessTokenResponse.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package org.cloudfoundry.client.lib.oauth2;
+
+public class Oauth2AccessTokenResponse {
```
[the remark this answers is not in the document]
Comment (IvanBorislavovDimitrov) [reply cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/oauth2/Oauth2AccessTokenResponse.java:3]: We do not use immutables in this project

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java (hunk 1 above)
Comment (IvanBorislavovDimitrov) [reply cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java]: All the code in the java client is "dead code", only the interface is used

**Review by theghost5800 — APPROVED** [review]

**Pull request merged**